File: leetcode/template/0301-0400/0397.integer-replacement.js
/**
 * https://leetcode-cn.com/problems/integer-replacement/
 * 0397.整数替换
 * 难度：中等
 * 
 * 给定一个正整数 n，你可以做如下操作：
 * 1. 如果 n 是偶数，则用 n / 2替换 n。
 * 2. 如果 n 是奇数，则可以用 n + 1或n - 1替换 n。
 * n 变为 1 所需的最小替换次数是多少？
 * 
 * 示例 1：
 * 输入：
 * 8
 * 
 * 输出：
 * 3
 * 
 * 解释：
 * 8 -> 4 -> 2 -> 1
 * 
 * 示例 2：
 * 输入：
 * 7
 * 
 * 输出：
 * 4
 * 
 * 解释：
 * 7 -> 8 -> 4 -> 2 -> 1
 * 或
 * 7 -> 6 -> 3 -> 2 -> 1
 */

/**
 * @param {number} n
 * @return {number}
 */
var integerReplacement = function(n) {
    
};

// Local test
let assert = require("assert");
console.time("Time cost");

assert.deepEqual(integerReplacement("param"), "expect", "caseName");
assert.deepEqual(integerReplacement("param"), "expect", "caseName");

console.log("Good job! We have passed all test case.");
console.timeEnd("Time cost");

File: leetcode/template/0301-0400/0397.integer-replacement.py
# https://leetcode-cn.com/problems/integer-replacement/
# 0397.整数替换
# 难度：中等
# 
# 给定一个正整数 n，你可以做如下操作：
# 1. 如果 n 是偶数，则用 n / 2替换 n。
# 2. 如果 n 是奇数，则可以用 n + 1或n - 1替换 n。
# n 变为 1 所需的最小替换次数是多少？
# 
# 示例 1：
# 输入：
# 8
# 
# 输出：
# 3
# 
# 解释：
# 8 -> 4 -> 2 -> 1
# 
# 示例 2：
# 输入：
# 7
# 
# 输出：
# 4
# 
# 解释：
# 7 -> 8 -> 4 -> 2 -> 1
# 或
# 7 -> 6 -> 3 -> 2 -> 1


class Solution:
    def integerReplacement(self, n: int) -> int:
        "put solution here"


# Local test
if __name__ == '__main__':
    import time
    launch_start = time.time()
    unit = Solution()

    assert unit.integerReplacement('param') == 'expect', 'Case 1'
    assert unit.integerReplacement('param') == 'expect', 'Case 2'

    print('Good job! We have passed all test case.')
    print('Time cast: ' + str(int((time.time() - launch_start) * 1000000)/1000) + 'ms')
